src/index.js

```diff
@@ -0,0 +1,85 @@
+const _ = require('lodash');
+const babel = require('rollup-plugin-babel');
+const commonjs = require('rollup-plugin-commonjs');
+const fs = require('fs');
+const html = require('rollup-plugin-html');
+const minify = require('rollup-plugin-babel-minify');
+const name = require('require-package-name');
+const path = require('path');
+const postcss = require('rollup-plugin-postcss');
+const resolve = require('rollup-plugin-node-resolve');
+const config = require('./config');
+const translations = require('./plugins/translations-import');
+const { uiRouterTranslations, uiRouterTranslationsUMD } = require('./plugins/ui-router-translations');
+
+const getPlugins = pluginOptions => [
+  html(),
+  postcss(),
+  resolve(),
+  commonjs(),
+  translations({ translationsRoot: pluginOptions.translationsRoot }),
+  pluginOptions.isUMD
+    ? uiRouterTranslationsUMD({ translationsRoot: pluginOptions.translationsRoot })
+    : uiRouterTranslations({ translationsRoot: pluginOptions.translationsRoot }),
+  babel({
+    babelrc: false,
+    exclude: 'node_modules/**',
+    plugins: [
+      '@babel/plugin-syntax-dynamic-import',
+      'babel-plugin-angularjs-annotate',
+    ],
+  }),
+  pluginOptions.isUMD && pluginOptions.minify ? minify({ comments: false }) : null,
+];
+
+module.exports = (opts) => {
+  const workingDir = process.cwd();
+  const packageConfig = JSON.parse(fs.readFileSync(path.resolve(workingDir, 'package.json')));
+  const peerDeps = Object.keys(packageConfig.peerDependencies || {});
+  let { translationsRoot } = opts;
+
+  const distFileName = opts.dist ? opts.dist.filename : `${name.base(packageConfig.name) || path.basename(workingDir)}`;
+
+  if (!translationsRoot) {
+    translationsRoot = fs.existsSync(path.resolve(process.cwd(), 'src'))
+      ? path.resolve(process.cwd(), 'src')
+      : process.cwd();
+  }
+
+  const targets = [{
+    input: opts.input,
+    external: peerDeps.concat(opts.external || []),
+    output: [{
+      file: path.join(config.dist.path.module, `${distFileName}.js`),
+      format: 'es',
+    }],
+    plugins: getPlugins({ translationsRoot, isUMD: false }),
+  }];
+
+  if (opts.umd) {
+    targets.push({
```

Comment:
You should create only one object and push it 2 times, with the diff

Something like:

```javascript
const myTarget = {
  input: opts.input,
  external: peerDeps.concat(opts.external || []),
  output: [{
    file: path.join(config.dist.path.umd, `${distFileName}.min.js`),
    name: distFileName,
    format: 'umd',
    globals: _.zipObject(peerDeps, peerDeps),
  }],
  plugins: getPlugins({ translationsRoot, isUMD: true }),
}

target.push(myTarget); // Push the initial target
target.push(Object.assign(myTarget, { plugins: getPlugins({ translationsRoot, isUMD: true, minify: true }) }); // Push the min version

```

We avoid to maintain 2 objects :)